internal/powerplatform/api/api_client.go

```diff
@@ -37,27 +37,23 @@ func NewApiClientBase(config *config.ProviderConfig, baseAuth *Auth) *ApiClient
 	}
 }
 
-func TryGetScopeFromURL(url string) (string, error) {
+func TryGetScopeFromURL(url string, cloudConfig config.ProviderConfigUrls) (string, error) {
 	switch {
-	case strings.LastIndex(url, "api.bap.microsoft.com") != -1,
-		strings.LastIndex(url, "api.powerapps.com") != -1:
+	case strings.LastIndex(url, cloudConfig.BapiUrl) != -1,
+		strings.LastIndex(url, cloudConfig.PowerAppsUrl) != -1:
+		return cloudConfig.PowerAppsScope, nil
 
-		return "https://service.powerapps.com/.default", nil
-	case strings.LastIndex(url, "api.powerplatform.com") != -1:
+	case strings.LastIndex(url, cloudConfig.PowerPlatformUrl) != -1:
+		return cloudConfig.PowerPlatformScope, nil
 
-		return "https://api.powerplatform.com/.default", nil
-	case strings.LastIndex(url, ".com/") != -1:
-
-		scope := strings.SplitAfterN(url, ".com/", 2)[0]
-		scope = scope + ".default"
-		return scope, nil
 	default:
-		return "", errors.New("Unable to determine scope from url: '" + url + "'. Please provide your own scope.")
+		u, err := neturl.Parse(url)
+		return u.Scheme + "://" + u.Host + "/.default", err
```

Comment:
No sure, if this should not throw an error already here. This was done to explicitly catch all url/scopes that we don't expect or build or they simply don't exist in a form that we assume. 
If now you will build a scope with .../.default from it, you should be sure that if that scope is not correct/does not work, it should explicitly say so and not (for example) "could not login ...."

Reply:
I don't understand this comment.  I think the point of this case is when we're connecting to dataverse webapi.  Since that's not a fixed url.  Are there other use cases?